### PR TITLE
Map hostnames

### PR DESCRIPTION
**Pull request by MateoLostanlen**
We needed to identify the device that sends the image to the central pi. We assigned a hostname to each device. It was therefore necessary to do a mapping between the ip that we retrieve with the image and this hostname. This mapping does not work inside the docker. My proposal is to mount a volume with the docker and to put the mapping json in this volume. Then you just have to launch a cron job with the scipt ip_hostname_mapping.py to update this mapping regularly. The web server inside the docker will then have access via the volume to an updated mapping

### CODE REVIEW

File: server/ip_hostname_mapping.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import json
+import socket
```
Comment (fe51) [inline server/ip_hostname_mapping.py:2]: Shouldn't we add `socket` to the docker requirement ?

File: server/src/app/api/routes/inference.py [hunk 2]
```diff
@@ -46,8 +47,13 @@ def predict_and_alert(file, ip):
     # Load Image
     image = Image.open(io.BytesIO(file))
 
+    # Get hosname
```
Comment (fe51) [inline server/src/app/api/routes/inference.py:50]: Just typo
```suggestion
    # Get hostname
```

File: server/ip_hostname_mapping.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+
+with open('data/ip_hostname_mapping.json') as f:
+    ip_hostname_mapping = json.load(f)
+
+inv_map = {v: k for k, v in ip_hostname_mapping.items()}
+ip_hostname_mapping = {}
+hostnames = inv_map.keys()
```
Comment (fe51) [inline server/ip_hostname_mapping.py:10]: style : maybe some blank lines here ?

**Review by fe51 — COMMENTED** [review]
Hi there ! Thanks for the PR ! 

Just some style and requirement comment. 

I have tested the `ip_hostname_mapping.py`.

Can you state yo have tested on the docker running on pi ?

File: server/ip_hostname_mapping.py (hunk 1 above)
Comment (MateoLostanlen) [reply server/ip_hostname_mapping.py:2]: This should be lunch by a cron jon outside docker. Moreover socket is pre-installed :)

File: server/src/app/api/routes/inference.py (hunk 2 above)
Comment (MateoLostanlen) [reply server/src/app/api/routes/inference.py]: thanks

File: server/ip_hostname_mapping.py (hunk 3 above)
Comment (MateoLostanlen) [reply server/ip_hostname_mapping.py:9]: ok :)

**Comment by MateoLostanlen** [comment]
> 
> 
> Hi there ! Thanks for the PR !
> 
> Just some style and requirement comment.
> 
> I have tested the `ip_hostname_mapping.py`.
> 
> Can you state yo have tested on the docker running on pi ?

Hi @fe51, thanks for the review, yes it has been tested on my test tower !

File: server/ip_hostname_mapping.py (hunk 1 above)
Comment (fe51) [reply server/ip_hostname_mapping.py:2]: Indeed ! Double indeed :)

However, if this job is not used by the wodker/webserver, shoulnd't it be in pyroengine/pi_utils ?

Comment (MateoLostanlen) [reply server/ip_hostname_mapping.py]: ok, I've made the change :)

Comment (fe51) [reply server/ip_hostname_mapping.py]: Perfect, thanks a lot :)

**Review by fe51 — APPROVED** [review]

**Pull request merged**